src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,348 @@
+        originalContextId: "",
+        contextId: "",
+        contextIdErrors: "",
+        contextName: "",
+        contextNameErrors: "",
+        flatPrefs: {} // For use in prefsetExists
+    },
+    selectors: {
+        contextIdInput: "#pmt-edit-context-id-input",
+        contextNameInput: "#pmt-edit-context-name-input"
+    },
+    bindings: {
+        contextIdInput: "contextId",
+        contextNameInput: "contextName"
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.editContextDialog.acceptConfirmDialog",
+            args: ["{that}.model", "{that}.applier", "{that}.closeDialog", "{arguments}.0"]
+        }
+    },
+    modelListeners: {
+        "contextIdErrors": {
+            func: "{that}.reRender"
+        },
+        "contextNameErrors": {
+            func: "{that}.reRender"
+        }
+    }
+});
+
+gpii.devpmt.dialogs.editContextDialog.acceptConfirmDialog = function (model, applier, closeDialog, event) {
+    // We don't want the form to actually submit the page, just to enable
+    // activating the submit button on Enter
+    event.preventDefault();
+
+    // Validate context name, eventually this will be replaced with
+    // our new json schema work.
+    if (model.contextId === "") {
+        applier.change("contextIdErrors", "Please enter an ID.");
+        return;
+    }
+    // TODO standardize the accepted regular expression for prefset ID's
+    // and other parts of the schema.
+    else if (!/^[\w-]+$/.exec(model.contextId)) {
+        applier.change("contextIdErrors", "Please use only alphanumeric characters.");
+        return;
+    }
+    // If we changed the contextId, make sure that there isn't an existing context with
+    // that ID
+    else if (model.contextId !== model.originalContextId &&
+             gpii.devpmt.prefsetExists(model.flatPrefs, model.contextId)) {
+        applier.change("contextIdErrors", "A Preference Set with that ID already exists.");
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        applier.change("contextIdErrors", "");
+    }
+
+    if (model.contextName === "") {
+        applier.change("contextNameErrors", "Please enter a name.");
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        applier.change("contextNameErrors", "");
+    }
+
+    var transaction = applier.initiate();
+    var prefsetToEdit = fluid.copy(model.flatPrefs.contexts[model.originalContextId]);
+    prefsetToEdit.name = model.contextName;
+    var oldPath = "flatPrefs.contexts." + model.originalContextId;
+    var path = "flatPrefs.contexts." + model.contextId;
+
+    transaction.fireChangeRequest({
+        path: oldPath,
+        type: "DELETE"
+    });
+
+    transaction.fireChangeRequest({
+        path: path,
+        value: prefsetToEdit
+    });
+
+    transaction.commit();
```

Comment:
Yeah yeah yeah good!

Reply:
Thanks... I assume thsi means it's ok. :p